MovieCatalogCrawler/src/main/java/moviecrawler/schedulers/Scheduler.java

```diff
@@ -0,0 +1,39 @@
+package moviecrawler.schedulers;
+
+import moviecrawler.URLEnums;
+import moviecrawler.crawlers.BasicWebCrawler;
+import moviecrawler.daos.MovieDAO;
+import moviecrawler.dtos.MovieDTO;
+import moviecrawler.entities.Movie;
+import moviecrawler.repositories.MovieRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Component
+public class Scheduler {
+
+    @Autowired
+    MovieRepository movieRepository;
+
+    @Scheduled(fixedRate = 3_600_000)
+    public void populateMovies() throws IOException {
+        for (URLEnums url : URLEnums.values()) {
+            final List<MovieDTO> pageLinks = new BasicWebCrawler().getMovieTitles(url.getValue());
+            Set<Movie> moviesInDatabase = new HashSet<Movie>(movieRepository.findAll());
+
+            for (MovieDTO movieDTO : pageLinks) {
```

Comment:
This can be further refactored if we load all existing movies in a hashmap on application start. Then we can filter results the same way against the hashmap but we will avoid further db calls on every iteration. This will require syncing the hasmap against the db.

Reply:
Repositories return list. If you want to create a repository by yourself and not use Spring, be my guest.